src/Sentry.OpenTelemetry/SentrySpanProcessor.cs

```diff
@@ -286,36 +295,57 @@ private static SpanStatus GetErrorSpanStatus(IDictionary<string, object?> attrib
-    //
-    //         var sentryEvent = new SentryEvent(exception, @event.Timestamp);
-    //         _hub.CaptureEvent(sentryEvent, scope =>
-    //         {
-    //             scope.Contexts["otel"] = GetOtelContext(spanAttributes);
-    //
-    //             var trace = scope.Contexts.Trace;
-    //             trace.TraceId = activity.TraceId.AsSentryId();
-    //             trace.SpanId = activity.SpanId.AsSentrySpanId();
-    //             trace.ParentSpanId = activity.ParentSpanId.AsSentrySpanId();
-    //         });
-    //     }
+        // https://develop.sentry.dev/sdk/performance/opentelemetry/#step-7-define-generatesentryerrorsfromotelspan
+        // https://opentelemetry.io/docs/specs/otel/trace/semantic_conventions/exceptions/
+        foreach (var @event in activity.Events.Where(e => e.Name == SemanticConventions.AttributeExceptionEventName))
+        {
+            var eventAttributes = @event.Tags.ToDictionary();
+            // This would be where we would ideally implement full exception capture. That's not possible at the
+            // moment since the full exception isn't yet available via the OpenTelemetry API.
+            // See https://github.com/open-telemetry/opentelemetry-dotnet/issues/2439#issuecomment-1577314568
+            // if (!eventAttributes.TryGetTypedValue("exception", out Exception exception))
+            // {
+            //      continue;
+            // }
+
+            // At the moment, OTEL only gives us `exception.type`, `exception.message`, and `exception.stacktrace`...
+            // So the best we can do is a poor man's exception (no accurate symbolication or anything)
+            if (!eventAttributes.TryGetTypedValue(SemanticConventions.AttributeExceptionType, out string exceptionType))
+            {
+                continue;
+            }
+            eventAttributes.TryGetTypedValue(SemanticConventions.AttributeExceptionMessage, out string message);
+            eventAttributes.TryGetTypedValue(SemanticConventions.AttributeExceptionStacktrace, out string stackTrace);
+
+            Exception exception;
+            try
+            {
+                var type = Type.GetType(exceptionType)!;
+                exception = (Exception)Activator.CreateInstance(type, message)!;
+                exception.SetSentryMechanism("SentrySpanProcessor.ErrorSpan");
+            }
+            catch
+            {
+                _options?.DiagnosticLogger?.LogError($"Failed to create poor man's exception for type : {exceptionType}");
+                continue;
+            }
+
+            // TODO: Validate that our `DuplicateEventDetectionEventProcessor` prevents this from doubling exceptions
+            // that are also caught by other means, such as our AspNetCore middleware, etc.
+            // (When options.RecordException = true is set on AddAspNetCoreInstrumentation...)
+            // Also, in such cases - how will we get the otel scope and trace context on the other one?
```

Comment:
The PR looks good to me. The TODO here sounds like quite the gotcha.

Reply:
It does. It occurs if you do something like this:
```csharp
        using var activity = Telemetry.ActivitySource.StartActivity("Funny business");
        try
        {
            throw new Exception("test activity.RecordException");
        }
        catch (Exception e)
        {
            SentrySdk.CaptureException(e);
            activity?.RecordException(e);
            activity?.SetStatus(Status.Error.WithDescription(e.Message));
        }
        return "Catch";
```

The deduplication doesn't work at all (since it matches by reference). Again, we could come up with convoluted ways of working around this but I think the ultimate solution is to address the issue in the OpenTelemetry codebase rather than try to work around it in ours. 

I'll create an issue for the above in our repository saying as much though.

Reply:
https://github.com/getsentry/sentry-dotnet/issues/2519